src/haz3lcore/statics/Term.re

```diff
@@ -128,6 +200,23 @@ module UTyp = {
       | Sum(uts) => Sum(to_ctr_map(ctx, uts))
       | List(u) => List(to_typ(ctx, u))
       | Parens(u) => to_typ(ctx, u)
+      | Forall({term: Var(name), _} as utpat, tbody) =>
+        let ctx =
+          Ctx.extend_tvar(
+            ctx,
+            {name, id: UTPat.rep_id(utpat), kind: Abstract},
+          );
+        Forall(name, to_typ(ctx, tbody));
+      | Forall(_, tbody) => to_typ(ctx, tbody)
```

Comment:
* don't use underscore here in case of future expansion of UTPat
* if there is a UTPat hole here, then I think it should still elaborate to a forall -- we can either generate a fresh variable, or preferably add UTPat holes into the Typ representation as well. It should behave just like a variable that is never used.

Reply:
Do you just mean give the variable a name? I think the linter complains of it being an unused variable since we don't use it in the case where it's not a type variable name.